internal/kubernetes/utils.go

```diff
@@ -248,6 +278,28 @@ func mergeMapsByPrefix(newMap map[string]string, oldMap map[string]string, prefi
 	return newMap
 }
 
+func mergeFinalizers(new *metav1.ObjectMeta, old metav1.ObjectMeta) {
+	new.SetFinalizers(mergeSlices(new.Finalizers, old.Finalizers))
+}
+
+func mergeSlices(newSlice []string, oldSlice []string) []string {
```

Comment:
Maybe we can simplify this a bit. The [controllerutil package](https://github.com/kubernetes-sigs/controller-runtime/blob/30dea34848b0e0b30690dd461a0eadb74d4869f3/pkg/controller/controllerutil/controllerutil.go#L353) already contains a helper that checks for redundant finalizers. Just use the old finalizers slice to preserve the order and add each of the new finalizers.